### PR TITLE
[17.0][PORT] account_invoice_start_end_dates: add start/end dates in tree view of move lines

### PR DESCRIPTION
**Pull request by henrybackman**
Ported this from 14.0 to 17.0: 
- [https://github.com/OCA/account-closing/pull/195](https://github.com/OCA/account-closing/pull/195)

### CODE REVIEW

File: account_invoice_start_end_dates/views/account_move_line.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <record id="view_move_line_form" model="ir.ui.view">
+        <field name="name">start_end_dates.view_move_line_form</field>
+        <field name="model">account.move.line</field>
+        <field name="inherit_id" ref="account.view_move_line_form" />
+        <field name="arch" type="xml">
+            <field name="date_maturity" position="after">
+                <field name="must_have_dates" invisible="1" />
+                <field
+                    name="start_date"
+                    attrs="{'required': [('must_have_dates', '=', True)]}"
```
Comment (sebalix) [inline account_invoice_start_end_dates/views/account_move_line.xml]:
```suggestion
                    required="must_have_dates"
```

File: account_invoice_start_end_dates/views/account_move_line.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                />
+                <field
+                    name="end_date"
+                    attrs="{'required': [('must_have_dates', '=', True)]}"
```
Comment (sebalix) [inline account_invoice_start_end_dates/views/account_move_line.xml]:
```suggestion
                    required="must_have_dates"
```

File: account_invoice_start_end_dates/views/account_move_line.xml (hunk 1 above)
Comment (henrybackman) [reply account_invoice_start_end_dates/views/account_move_line.xml]: Fixed

File: account_invoice_start_end_dates/views/account_move_line.xml (hunk 2 above)
Comment (henrybackman) [reply account_invoice_start_end_dates/views/account_move_line.xml]: Fixed

**Comment by sebalix** [comment]
In fact the change was already there but done through another commit. The only relevant change brought here is the split of XML views in two files: `account_move.xml` and `account_move_line.xml`, reducing the gap of commits between 14.0 and 17.0.

**Review by sebalix — APPROVED** [review]

**Review by gbrito — APPROVED** [review]

**Comment by henrybackman** [comment]
@sebalix The runboat/build check seems stuck. Is there a way to restart that?

**Pull request merged**